router/http/src/main/scala/io/buoyant/router/http/ApplyHostForwardedHeader.scala

```diff
@@ -0,0 +1,65 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Fields, Request, Response, HeaderMap}
+import com.twitter.util.Future
+import java.net.{Inet4Address, Inet6Address, InetSocketAddress, SocketAddress, URI, URISyntaxException}
+import scala.collection.mutable
+import scala.util.Random
+import com.twitter.logging.Logger
+
+/**
+ * Applies the Host value in the [Forwarded](https://tools.ietf.org/html/rfc7239) header to the Request's Host header.
+ * So the Request is sent with the actual host value.
+ */
+class ApplyHostForwardedHeader() extends SimpleFilter[Request, Response] {
+  private val log = Logger(getClass)
+
+  val forwardedHeader = "Forwarded"
+  def apply(req: Request, svc: Service[Request, Response]): Future[Response] = {
+
+    replaceHostWithForwardedHostIfExists(req)
+
+    svc(req)
+  }
+
+  private def getForwardedHost(headerMap: HeaderMap): Option[String] = {
```

Comment:
Should we support comma separated fields too?

From the RFC:

> Examples:
> 
>        Forwarded: for="_gazonk"
>        Forwarded: For="[2001:db8:cafe::17]:4711"
>        Forwarded: for=192.0.2.60;proto=http;by=203.0.113.43
>        Forwarded: for=192.0.2.43, for=198.51.100.17
> 
>    Note that as ":" and "[]" are not valid characters in "token", IPv6
>    addresses are written as "quoted-string".
> 
>    A proxy server that wants to add a new "Forwarded" header field value
>    can either append it to the last existing "Forwarded" header field
>    after a comma separator or add a new field at the end of the header
>    block.

Reply:
Done